packages/runner-ct/src/iframe/iframes.tsx

```diff
@@ -22,134 +21,133 @@ export function getSpecUrl ({ namespace, spec }, prefix = '') {
 
-    this.iframeModel = new IframeModel({
-      state: this.props.state,
-      restoreDom: this.autIframe.restoreDom,
-      highlightEl: this.autIframe.highlightEl,
-      detachDom: this.autIframe.detachDom,
-      snapshotControls: (snapshotProps) => (
-        <SnapshotControls
-          eventManager={this.props.eventManager}
-          snapshotProps={snapshotProps}
-          state={this.props.state}
-          onToggleHighlights={this._toggleSnapshotHighlights}
-          onStateChange={this._changeSnapshotState}
-        />
-      ),
-    })
+      autIframe.highlightEl(snapshot, snapshotProps)
+    } else {
+      autIframe.removeHighlights()
+    }
+  }
 
-    this.iframeModel.listen()
+  const _changeSnapshotState = (snapshotProps, index) => {
+    const snapshot = snapshotProps.snapshots[index]
 
-    this._disposers.push(autorun(() => {
-      const spec = this.props.state.spec
+    props.state.snapshot.stateIndex = index
+    autIframe.restoreDom(snapshot)
 
-      if (spec) {
-        this._run(spec, config)
-      }
-    }))
+    if (props.state.snapshot.showingHighlights && snapshotProps.$el) {
+      autIframe.highlightEl(snapshot, snapshotProps)
+    } else {
+      autIframe.removeHighlights()
+    }
   }
 
-  @action _setScriptError = (err: string | undefined) => {
-    this.props.state.scriptError = err
-  }
+  const _setScriptError = action((err: string | undefined) => {
+    props.state.scriptError = err
+  })
 
-  _run = (spec, config) => {
+  const _run = (spec, config) => {
     config.spec = spec
 
-    // this.props.eventManager.notifyRunningSpec(specPath)
-    // logger.clearLog()
-    this._setScriptError(undefined)
+    _setScriptError(undefined)
 
-    this.props.eventManager.setup(config)
+    props.eventManager.setup(config)
 
     // This is extremely required to not run test till devtools registered
-    when(() => this.props.state.readyToRunTests, () => {
-      window.Cypress.on('window:before:load', this.props.state.registerDevtools)
+    when(() => props.state.readyToRunTests, () => {
+      window.Cypress.on('window:before:load', props.state.registerDevtools)
 
-      const $autIframe = this._loadIframes(spec)
+      const $autIframe = _loadIframes(spec)
 
-      this.props.eventManager.initialize($autIframe, config)
+      props.eventManager.initialize($autIframe, config)
     })
   }
 
   // jQuery is a better fit for managing these iframes, since they need to get
   // wiped out and reset on re-runs and the snapshots are from dom we don't control
-  _loadIframes (spec) {
-    const specSrc = getSpecUrl({ namespace: this.props.config.namespace, spec })
-    const $container = $Cypress.$(this.containerRef).empty()
-    const $autIframe = this.autIframe.create().appendTo($container)
+  const _loadIframes = (spec: Cypress.Cypress['spec']) => {
+    const specSrc = getSpecUrl({ namespace: props.config.namespace, spec })
+    const $container = $Cypress.$(containerRef.current).empty()
```

Comment:
You might have to handle `containerRef.current === null` here

Reply:
I just added a defensive check. Not sure if it can even be null at the time this code runs, but better safe than sorry.